crates/turbopack/src/lib.rs

```diff
@@ -146,10 +147,28 @@ async fn apply_module_type(
         ModuleType::Json => JsonModuleAssetVc::new(source).into(),
         ModuleType::Raw => source,
         ModuleType::Css(transforms) => {
-            CssModuleAssetVc::new(source, context.into(), *transforms).into()
+            if matches!(
+                &*intention,
+                EnvironmentIntention::Api
+                    | EnvironmentIntention::Middleware
+                    | EnvironmentIntention::Data
+            ) {
```

Comment:
Perhaps this should be a method on `environment` to check whether a particular module type is supported or not. I'm wary of this match not being exhaustive in the future if we add another environment intention, or custom environments.